discovery/kubernetes/kubernetes.go

```diff
@@ -222,6 +222,10 @@ func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 				break
 			}
 		}
+		// The pod role supports node selectors when configured with `attach_metadata: {node: true}`.
```

Comment:
Do we really need to fail here? It might be simpler to add `RoleNode` to `allowedSelectors`  for `RolePod`.

Reply:
I think it's better to "early/explicitly" fail if the node selectors have no effect.

Reply:
In that case, the error message should be more explicit. I'd rather add RoleNode to the list of allowed selectors for RolePod and have an additional check here.
```suggestion
		// The pod role supports node selectors only when configured with `attach_metadata: {node: true}`.
		if selector.Role == RoleNode && c.Role == RolePod && !c.AttachMetadata.Node {
			return fmt.Errorf("pod role with node selector requires node metadata to be enabled")
		}
```